ostd/src/sync/wait.rs

```diff
@@ -65,53 +65,39 @@ impl WaitQueue {
     ///
     /// By taking a condition closure, this wait-wakeup mechanism becomes
     /// more efficient and robust.
-    pub fn wait_until<F, R>(&self, mut cond: F) -> R
+    pub fn wait_until<F, R>(&self, cond: F) -> R
     where
         F: FnMut() -> Option<R>,
     {
-        if let Some(res) = cond() {
-            return res;
-        }
-
-        let (waiter, _) = Waiter::new_pair();
-
-        self.wait_until_or_cancelled(cond, waiter, || false)
+        self.wait_until_or_cancelled(cond, || Ok::<(), ()>(()))
             .unwrap()
     }
 
     /// Waits until some condition is met or the cancel condition becomes true.
     ///
-    /// This method will return `Some(_)` if the condition returns `Some(_)`, and will return
-    /// the condition test result regardless what it is when the cancel condition becomes true.
-    #[doc(hidden)]
-    pub fn wait_until_or_cancelled<F, R, FCancel>(
+    /// This method will return `Ok(_)` if the condition returns `Some(_)`, and will stop waiting
+    /// if the cancel condition returns `Err(_)`. In this situation, this method will return the `Err(_)`
+    /// generated by the cancel condition.
+    pub fn wait_until_or_cancelled<F, R, FCancel, E>(
```

Comment:
I think this method should be removed.

The current call chain is like this:
```
                WaitQueue::wait_until -> Waiter::wait_until
                                                 ^
                                                 |
                                         Waiter::wait_until_or_cancelled
                                                 ^
                                                 |                           (in OSTD)
=================================================|=====================================
                                                 |                           (in Asterinas)
                                                 |
    WaitQueue::wait_until_or_timeout -> Waiter::wait_until_or_timeout[_cancelled]
                                                 ^
                                                 |
   WaitQueue::pause_until_or_timeout -> Waiter::pause_until_or_timeout
```

So this method is not useful. This method was previously marked as `#[doc(hidden)]`, so it should be removed now.